main.py

```diff
@@ -104,33 +104,31 @@ def get_s3_keys(bucket, prefix):
 
 
 def produce_kafka_messages(bucket, job_id, fixture_data, args):
-    # Process each fixture data dir
+    # Process each fixture data dir, sending each file in it to kafka as a payload
     producer = KafkaProducer(
         bootstrap_servers=args.kafka_bootstrap_servers,
         ssl_check_hostname=args.ssl_broker,
     )
     s3_client = boto3.client("s3")
     for s3_key in fixture_data:
-        logger.debug(f"Processing key: {s3_key}")
-        line_no = 0
-        for line in (
-            s3_client.get_object(Bucket=bucket, Key=s3_key)["Body"]
-            .read()
-            .splitlines()
-        ):
-            line_no += 1
-            try:
-                data = json.loads(line)
-                db = data["message"]["db"] if "db" in data["message"] else "missingDb"
-                collection = data["message"]["collection"] if "collection" in data["message"] else "missingCollection"
-                topic_name = f"{args.topic_prefix}{job_id}_{db}.{collection}"
-            except json.JSONDecodeError as err:
-                logger.error(
-                    f"line {line_no} of {s3_key} contains invalid JSON data: {err.msg}"
-                )
-                continue
-            producer.send(topic_name, line)
-            producer.flush()
+        logger.info(f"Processing key: {s3_key}")
+        payload = s3_client.get_object(Bucket=bucket, Key=s3_key)["Body"].read()
```

Comment:
What happens if this doesn't exist? Does the error just bubble up to the logs? If so, might be ok, just thinking of test debugging, file missing/not accessible would be a good explicit thing to see in the logs.

Reply:
Yes, it will blow up afaik